src/rgw/CMakeLists.txt

```diff
@@ -19,6 +19,8 @@ function(gperf_generate input output)
     )
 endfunction()
 
+add_definitions(-DBOOST_USE_VALGRIND)
+
```

Comment:
i think this one will need to apply globally in ceph, because we use boost all over. we don't want rgw to build it differently than other libraries like common/librados

it's also not clear what performance impact this has, so we might only want to turn it on when `ALLOCATOR=libc`, which (i think?) is the only way we can run with valgrind

i'd also like to see a short comment here describing why we need the flag

Reply:
It should be turned on when WITH_BOOST_VALGRIND is turned on. I thought it was.

Reply:
thanks for the pointer @adamemerson!

in CMakeLists.txt:
```cmake
CMAKE_DEPENDENT_OPTION(WITH_BOOST_VALGRIND "Boost support for valgrind" OFF
  "NOT WITH_SYSTEM_BOOST" OFF)
```
and cmake/modules/BuildBoost.cmake:
```cmake
    if((c MATCHES "coroutine|context") AND (WITH_BOOST_VALGRIND))
      set_target_properties(Boost::${c} PROPERTIES
        INTERFACE_COMPILE_DEFINITIONS "BOOST_USE_VALGRIND")
    endif()
```
it looks like that only takes effect for non-system boost, which i don't think is the case for any of our shaman builds. for our teuthology testing, we'd specifically need this on for the centos notcmalloc build